### PR TITLE
Vite: preview generator

### PR DESCRIPTION
**Pull request by vicb**
This PR is the second part of #14125

I have created multiple commits to make the review easier.

Commits (older first):

- [cleaning(vite): minor cleanup/refactoring](https://github.com/nrwl/nx/pull/14375/commits/6c06134c39e51e171c78005a0ad0e67046c7b61e)

A couple fixes (see inline comments) and some refactoring/simplification of the vite generators. The behavior remains unchanged.

- [fix(vite): fix addOrChangeServeTarget](https://github.com/nrwl/nx/pull/14375/commits/d3becb8ff193b151993c183e0812054667d0c6c9)

A small fix

- [feat(vite): generate a preview target](https://github.com/nrwl/nx/pull/14375/commits/865e7d0f625f27f3b03e5da5d3e0647c63bc48ab)

Update the generator to generate a preview target.

I tested the PR locally and it works fine.

The next and last step will be to add support for watch mode in the build and preview executors.

/cc @mandarini

### CODE REVIEW

File: packages/vite/src/utils/generator-utils.ts [hunk 1]
```diff
@@ -89,119 +69,59 @@ export function findExistingTargetsInProject(
...
-      unsupportedUserProvidedTargetServe = true;
-    }
-  }
+  checkUserProvidedTarget('build', supportedBuilders);
+  checkUserProvidedTarget('serve', supportedServers);
+  checkUserProvidedTarget('test', supportedTesters);
 
-  if (userProvidedTargets?.test) {
-    if (
-      arrayOfSupportedServers.includes(
```
Comment (vicb) [inline packages/vite/src/utils/generator-utils.ts:128]: `arrayOfSupportedServers` should have been `arrayOfSupportedTesters` here.
Fixed by the PR.

File: packages/vite/src/utils/generator-utils.ts [hunk 2]
```diff
@@ -756,11 +667,10 @@ function handleViteConfigFileExists(
       formats: ['es', 'cjs'],
     },
     rollupOptions: {
-      external: [
```
Comment (vicb) [inline packages/vite/src/utils/generator-utils.ts:759]: Here was the second errors.

The possible values were `["'react', 'react-dom', 'react/jsx-runtime'"] ` or `[]`.

They are now ['react', 'react-dom', 'react/jsx-runtime'] or `[]`.

**Review by mandarini — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

**Comment by mandarini** [comment]
Perfect, thank you @vicb ! :D Tested it, too, locally. Everything works as expected.

**Comment by vicb** [comment]
Thanks!

**Pull request merged**